### PR TITLE
Dartino debugging on device

### PR DESCRIPTION
**Pull request by danrubel**
First cut debugging support - not enabled by default
@devoncarew

### CODE REVIEW

File: lib/dartino/launch_dartino.dart [hunk 1]
```diff
@@ -113,4 +115,56 @@ class DartinoLaunch extends Launch {
...
+    runner.onStdout.listen((str) {
+      if (str.startsWith('localhost:')) {
+        try {
+          portCompleter.complete(int.parse(str.substring(10)));
+        } catch (e) {
+          pipeStdio('Failed to parse observatory port from "$str"',
+              error: true);
+        }
+      }
+      pipeStdio(str, subtle: true);
```
Comment (devoncarew) [inline lib/dartino/launch_dartino.dart]: You might want to move this above the if statement.

File: lib/dartino/launch_dartino.dart [hunk 2]
```diff
@@ -113,4 +115,56 @@ class DartinoLaunch extends Launch {
...
+    });
+    runner.onStderr.listen((str) => pipeStdio('\n$str\n', error: true));
+    runner.execStreaming().then((int exitCode) {
+      pipeStdio('debug session exit code is $exitCode', highlight: true);
+      launchTerminated(exitCode, quiet: true);
+    });
+    int observatoryPort = await portCompleter.future
+        .timeout(new Duration(seconds: 5), onTimeout: () => null);
```
Comment (devoncarew) [inline lib/dartino/launch_dartino.dart]: Does a 5 sec timeout work for most dartino launches?

**Comment by devoncarew** [comment]
Lgtm! Exciting to see it hooked it.

File: lib/dartino/launch_dartino.dart (hunk 1 above)
Comment (danrubel) [inline lib/dartino/launch_dartino.dart]: Good point. Done.

File: lib/dartino/launch_dartino.dart (hunk 2 above)
Comment (danrubel) [inline lib/dartino/launch_dartino.dart]: Don't know yet. I picked 5 seconds as a way to exit out of the process if the port is not found in a reasonable amount of time, but perhaps I could ask the user if they want to continue waiting or ? I'll add a note here to ponder and follow up with a better solution.

**Comment by danrubel** [comment]
Merged https://github.com/dart-atom/dartlang/commit/6459d053b05cd2fc30078415951180a701ef9518

File: lib/dartino/launch_dartino.dart (hunk 2 above)
Comment (devoncarew) [inline lib/dartino/launch_dartino.dart]: Most flutter apps would start in less than 5 secs, but the outliers - ones  that start but start slowly - are more in the neighborhood of 20 secs. No idea what dartino apps look like :) well probably get a sense of that just by using it.

**Pull request closed without merging**